src/components/input/Button/Button.styles.js

```diff
@@ -1,65 +1,85 @@
+export const setButtonColour = (intent) => {
+  console.log(intent);
+  //const currentButtonColour = buttonColors[intent];
+  console.log(currentButtonColour);
+  return currentButtonColour;
+};
+
+export const buttonProminences = (intent) => {
   primary: css`
-    background-color: ${buttonColors.buttonPrimaryBackground};
-    border: 1px solid ${buttonColors.buttonPrimaryBorder};
-    color: ${buttonColors.buttonPrimaryText};
+    background-color: ${setButtonColour(intent)};
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.primaryText};
 
     &:hover {
-      background-color: ${buttonColors.buttonPrimaryBackgroundHover};
+      background-color: ${buttonColors.hover};
     }
   `,
-  secondary: css`
-    background-color: ${buttonColors.buttonSecondaryBackground};
-    border: 1px solid ${buttonColors.buttonSecondaryBorder};
-    color: ${buttonColors.buttonSecondaryText};
+  default: css`
+    background: none;
+    border: 2px solid ${buttonColors.default};
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonSecondaryBackgroundHover};
+      border-color: ${buttonColors.hover};
+      color: ${buttonColors.hover};
     }
   `,
-  danger: css`
-    background-color: ${buttonColors.buttonDangerBackground};
-    border: 1px solid ${buttonColors.buttonDangerBorder};
-    color: ${buttonColors.buttonDangerText};
+  minimal: css`
+    background: none;
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonDangerBackgroundHover};
+      color: ${buttonColors.hover};
     }
   `,
-  success: css`
-    background-color: ${buttonColors.buttonSuccessBackground};
-    border: 1px solid ${buttonColors.buttonSuccessBorder};
-    color: ${buttonColors.buttonSuccessText};
+};
 
-    &:hover {
-      background-color: ${buttonColors.buttonSuccessBackgroundHover};
-    }
+// success, warning, danger, none
+export const buttonIntents = {
+  success: css`
+    
+    }, 
   `,
 };
 
-export const buttonSizes = {
-  small: css`
-    ${fontSizes.size0};
-    padding: 8px 11px;
+
+// active, hover, focus, disabled
+export const buttonState = {
+  active: css`
   `,
-  medium: css`
-    ${fontSizes.size1};
-    padding: 10px 12px;
+};
+
+// navigaation or action
+export const buttonBehaviours = {
+  navigation: css`
+    &::after {
+      content: ' →';
+      display: inline;
+    }
   `,
-  large: css`
-    ${fontSizes.size2};
-    padding: 12px 15px;
+  action: css`
   `,
 };
 
+// Border-radius should be brandable!
 export const StyledButton = styled.button`
-  ${p => buttonModes[p.mode]};
-  ${p => buttonSizes[p.size]};
-  border-radius: 3px;
   cursor: pointer;
+  margin: 0 ${spacerSizes.xsmall} ${spacerSizes.small};
+  padding: ${spacerSizes.xsmall} ${spacerSizes.medium};
+  ${interfaceTypography.button};
+  ${(props, intent) => buttonProminences[props.prominence]};
```

Comment:
So `${foo}` in JavaScript is used to interpolate, eg to output a variable into a string easily. In here you are outputting a function that you are defining:

```js
(props, intent) => buttonProminences[props.prominence]
```

that defines a function that takes two arguments and outputs `buttonProminences[props.prominence]`. But you aren't _running_ it here; just defining it. It's the same as doing:

```js
const myFunction = (props, intent) => {
  return buttonProminences[props.prominence];
};
```

If you want to get the output of `(props, intent) => buttonProminences[props.prominence]` you'll need to write it as:

```js
${((props, intent) => buttonProminences[props.prominence])()}
```

buuuut I'm pretty sure that's not what you want here because it's got no arguments supplied to it. Let me look over your wall of text to see if I can make sure of what you're trying to do here.

Reply:
I'm super confused. So, what I'm trying to do is pass the intent value through to the function that defines & sets the CSS for buttons. This may not be possible within the existing structure, but I don't know how to make a new/better structure and I need some dev help. 

I cannot figure out how to do this from within the existing structure, and what I've pushed here has major issues that make things break. I can't figure out how to output the styles for the buttons, even with massive duplication. Could you maybe pull down the code, run it locally, and take a closer look at what's going on? I'm really stuck on this.

Reply:
What can I do to help explain the issue here better? Or would it make sense to nix the intent and state for buttons for now, and try that at a later date?